Source/Synchronization/Strategies/Asset Deletion/AssetDeletionRequestStrategy.swift

```diff
@@ -20,7 +20,18 @@ import Foundation
 
 fileprivate extension AssetRequestFactory {
     static func request(for identifier: String, on queue: ZMSGroupQueue, apiVersion: APIVersion, block: @escaping ZMCompletionHandlerBlock) -> ZMTransportRequest {
-        let request = ZMTransportRequest(path: "/assets/v3/\(identifier)", method: .methodDELETE, payload: nil, apiVersion: apiVersion.rawValue)
+
+        let path: String
+
+        switch apiVersion {
+        case .v0, .v1:
+            path = "/assets/v3/\(identifier)"
+        case .v2:
+            let domain = APIVersion.domain ?? ""
```

Comment:
An empty domain would result in a malformed endpoint path, so we shouldn't generate a request in this case.

```suggestion
            guard let domain = APIVersion.domain else { return nil }
```